rosidl_generator_py/resource/_msg_support.c.em

```diff
@@ -517,24 +517,44 @@ if isinstance(type_, AbstractNestedType):
     memcpy(dst, src, @(member.type.size) * sizeof(@primitive_msg_type_to_c(member.type.value_type)));
     Py_DECREF(field);
 @[    elif isinstance(member.type, AbstractSequence)]@
-    if (ros_message->@(member.name).size > 0) {
-      field = PyObject_GetAttrString(_pymessage, "@(member.name)");
-      if (!field) {
-        return NULL;
-      }
-      assert(field->ob_type != NULL);
-      assert(field->ob_type->tp_name != NULL);
-      assert(strcmp(field->ob_type->tp_name, "array.array") == 0);
-      // ensure that itemsize matches the sizeof of the ROS message field
-      PyObject * itemsize_attr = PyObject_GetAttrString(field, "itemsize");
-      assert(itemsize_attr != NULL);
-      size_t itemsize = PyLong_AsSize_t(itemsize_attr);
-      Py_DECREF(itemsize_attr);
-      if (itemsize != sizeof(@primitive_msg_type_to_c(member.type.value_type))) {
-        PyErr_SetString(PyExc_RuntimeError, "itemsize doesn't match expectation");
-        Py_DECREF(field);
-        return NULL;
+    field = PyObject_GetAttrString(_pymessage, "@(member.name)");
+    if (!field) {
+      return NULL;
+    }
+    assert(field->ob_type != NULL);
+    assert(field->ob_type->tp_name != NULL);
+    assert(strcmp(field->ob_type->tp_name, "array.array") == 0);
+    // ensure that itemsize matches the sizeof of the ROS message field
+    PyObject * itemsize_attr = PyObject_GetAttrString(field, "itemsize");
+    assert(itemsize_attr != NULL);
+    size_t itemsize = PyLong_AsSize_t(itemsize_attr);
+    Py_DECREF(itemsize_attr);
+    if (itemsize != sizeof(@primitive_msg_type_to_c(member.type.value_type))) {
+      PyErr_SetString(PyExc_RuntimeError, "itemsize doesn't match expectation");
+      Py_DECREF(field);
+      return NULL;
+    }
+    // clear the array, poor approach to remove potential default values
```

Comment:
We could also create a new array object with the same typecode. But I'm not sure if this would be any prettier, or if we still need to worry about decref'ing the individual elements.

In any case, this looks okay to me with green CI.

Reply:
I would be happy to do so but couldn't figure out how to instantiate an `array.array` from the C code when I wrote this originally :disappointed: